src/library/dao/trackdao.cpp

```diff
@@ -1227,7 +1227,22 @@ TrackPointer TrackDAO::getTrackFromDB(TrackId trackId) const {
         // Just to be safe, but this should never happen!!
         return pTrack;
     }
-    DEBUG_ASSERT(pTrack->getId() == trackId);
+    if (pTrack->getId() != trackId) {
+        // The cacheResolver() failed to resolve the track by it's trackId,
+        // but has found a different track that has already been loaded and
+        // is stored in the cache. That track references the same (physical)
+        // file on the file system!! Due to symbolic links different locations
+        // may resolve to the same canonical location. We can only load and
+        // access each file once at a time to prevent corruption due to
+        // concurrent, non-exclusive (write) access.
+        kLogger.warning()
+                << "Returning already loaded track"
+                << pTrack->getId()
+                << "instead of"
+                << trackId
+                << "with the same canonical file location"
+                << pTrack->getFileInfo().canonicalFilePath();
```

Comment:
I think we can safely return ```return TrackPointer();``` here. All using code need to handle that anyway. 

I have found three issue we need urgently fixed anyway: 

AutoDJFeature::slotAddRandomTrack() has a debug assert that is wrong

void TrackExportWorker::run(), CrateFeature::slotExportTrackFiles() and SetlogFeature::slotJoinWithPrevious() are also missing the null check.